### PR TITLE
rename options + add upgrades

### PR DESCRIPTION
**Pull request by vandenman**
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1941

### CODE REVIEW

File: inst/qml/RegressionLinearBayesian.qml [hunk 1]
```diff
@@ -263,43 +263,43 @@ Form {
...
-					value: "Bernoulli"; label: qsTr("Bernoulli")
+					value: "bernoulli"; label: qsTr("Bernoulli")
 					childrenOnSameRow: true
 					DoubleField { name: "bernoulliParam"; label: qsTr("p"); defaultValue: 0.5; max: 1; inclusive: JASP.None; decimals: 3 }
 				}
 			}
 
 			RadioButtonGroup
 			{
-				name: "samplingMethod"
```
Comment (vandenman) [inline inst/qml/RegressionLinearBayesian.qml:290]: I changed this because it was in the example, but I think `samplingMethod` is more clear than `sampling`. The former sounds like you need to specify a method (e.g., a string), while the latter sounds like you need to pass `TRUE` or `FALSE`.

**Comment by Kucharssim** [comment]
is this ready for review? :)

**Comment by vandenman** [comment]
It is now :) @Kucharssim I usually wait until the checks pass before assigning reviewers.

**Review by Kucharssim — CHANGES_REQUESTED** [review]
Most of the changes are good, just a couple of comments. Also, `priorRegressionCoefficients` options are not renamed but it seems that some of the names need attention

**Comment by vandenman** [comment]
Do you have any suggestions for `priorRegressionCoefficients`? I think that names like `AIC`, `BIC`, etc. make a lot of sense. Or do you mean `alpha` specifically? We could call that `priorHyperParameterValue` or so.

**Comment by Kucharssim** [comment]
Yeah sorry, I meant things like `EB-global` could be `ebGlobal`, and `rScale` and `alpha` could indeed be something like `gPriorAlpha`, `jzsRScale` to make it clear that these values belong to particular priors

**Comment by vandenman** [comment]
I'll change this because it's not so bad here. 

More generally I am wondering whether we can decrease the verbosity of this using nested lists.

Did you at one point consider doing something like [`DEoptim`](https://www.rdocumentation.org/packages/DEoptim/versions/2.2-6/topics/DEoptim) does with [`DEoptim.control`](https://www.rdocumentation.org/packages/DEoptim/versions/2.2-6/topics/DEoptim.control)? 

For example, in Descriptives the generated function could have a structure like so
```r
Descriptives <- function(
  variables, 
  <main arguments>, 
  statistics = DescriptivesStatistics(),
  customizablePlots = DescriptivesCustomizablePlots(),
  table = DescriptivesTable()
)

DescriptivesStatistics <- function(distributionPlot = FALSE, correlationPlot = FALSE, <more args>) {
  list(distributionPlot = distributionPlot, correlationPlot = correlationPlot, <more args>)
}
```

This could be done automatically for everything in a section (except for the first section) and optionally enabled in QML for other components (e.g., ones with a lot of nesting). The benefit of this this is that you can drop repeated parts of the option names making it ambiguous what names refer to (e.g., `distributionPlotBinwidthType` -> `binwidthType`). The reason for these helper functions with arguments as opposed to raw `list`s is that 1) there are still default arguments and 2) RStudio (and R) support autocomplete for it.

Do you think this makes sense? If yes, then perhaps we should continue this discussion in an issue :smile:

**Comment by Kucharssim** [comment]
I opened an issue for this

File: inst/Upgrades.qml [hunk 2]
```diff
@@ -117,6 +117,29 @@ Upgrades
...
+					case "BIC":									return "bic";
+					case "EB-global":							return "ebGlobal";
+					case "EB-local":							return "ebLocal";
+					case "g-prior":								return "gPrior";
+					case "hyper-g":								return "hyperG";
+					case "hyper-g-laplace""":					return "hyperGLaplace";
+					case "hyper-g-n":							return "hyperGN";
+					case "JZS":									return "jzs";
+				}
+			}
```
Comment (Kucharssim) [inline inst/Upgrades.qml:137]: This option does not get upgraded... I am starting to see a pattern here (https://github.com/jasp-stats/jaspBsts/pull/20#discussion_r938797655). Am I going crazy and there is some obvious mistake or this deserves an issue, @JorisGoosen?
[bayesian-regression_0.16.3.jasp.zip](https://github.com/jasp-stats/jaspRegression/files/9269093/bayesian-regression_0.16.3.jasp.zip)

Comment (vandenman) [reply inst/Upgrades.qml]: There is an obvious mistake:
```js
 					case "hyper-g-laplace""":					return "hyperGLaplace"; 
```
there should not be `"""` in `"hyper-g-laplace"""`.

@JorisGoosen the current behavior of the upgrader is to ignore the option without any error when this happens and I guess that goes for any error encountered in `jsFunction`s. Perhaps something to change alongside https://github.com/jasp-stats/jaspBsts/pull/20#discussion_r939611518?

File: inst/qml/RegressionLinearBayesian.qml [hunk 3]
```diff
@@ -263,43 +263,43 @@ Form {
...
+				name: "sampling"
```
Comment (Kucharssim) [inline inst/qml/RegressionLinearBayesian.qml:290]:
```suggestion
				name: "samplingMethod"
```
you can revert this change

**Review by Kucharssim — CHANGES_REQUESTED** [review]
Just one (hopefully last) thing

**Comment by JorisGoosen** [comment]
> I'll change this because it's not so bad here.
> 
> More generally I am wondering whether we can decrease the verbosity of this using nested lists.
> 
> Did you at one point consider doing something like [`DEoptim`](https://www.rdocumentation.org/packages/DEoptim/versions/2.2-6/topics/DEoptim) does with [`DEoptim.control`](https://www.rdocumentation.org/packages/DEoptim/versions/2.2-6/topics/DEoptim.control)?
> 
> ...
> 
> This could be done automatically for everything in a section (except for the first section) and optionally enabled in QML for other components (e.g., ones with a lot of nesting). The benefit of this this is that you can drop repeated parts of the option names making it ambiguous what names refer to (e.g., `distributionPlotBinwidthType` -> `binwidthType`). The reason for these helper functions with arguments as opposed to raw `list`s is that 1) there are still default arguments and 2) RStudio (and R) support autocomplete for it.
> 
> Do you think this makes sense? If yes, then perhaps we should continue this discussion in an issue 😄

This sounds like a good idea but would require a few changes in the backend of the qml-components (Im not sure in how many places we fully expect it to be flat list, I think a few). But it would make it a lot more readable and manageable.

File: inst/Upgrades.qml (hunk 2 above)
Comment (JorisGoosen) [reply inst/Upgrades.qml]: I responded there

**Review by Kucharssim — APPROVED** [review]
